### ISSUE TITLE
Ferocious Bite

### ISSUE THREAD

**Issue by Fragrus** [TrinityCore/TrinityCore#25743]
**Description:**

The spell does not apply to opponents

**Expected behaviour:**

Dealing damage

**Steps to reproduce the problem:**

1. Create druid and reach 7 level.
2. Use https://www.wowhead.com/spell=22568 and recieve error message: "Invalid target".

**Branch(es):**

master

**TC rev. hash/commit:** 

TrinityCore rev. ad0fa5ab4a36 2020-12-19 12:56:30 +1000 (master branch) (Unix, Release, Static)

**Operating system:** Debian 10

**Comment by matanshukry** [comment]
Seems to be due to this case:
https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Spells/Spell.cpp#L5380

In the type of Ferocious Bite, POWER_BURN is used with target (TARGET_UNIT_CASTER, NO_TARGET).
I wonder if a specific test for this would be enough; e.g.
```
if (effect->TargetA.GetTarget() != TARGET_UNIT_CASTER || effect->TargetB.GetTarget() != 0)
  if (m_caster->GetTypeId() == TYPEID_PLAYER)
    ...
```

@Shauren

**Comment by matanshukry** [comment]
Fixed by #26093
